### ISSUE TITLE
Loyalty Minimum spend not been effected for the tiers

### ISSUE THREAD

**Issue by olamide2** [frappe/erpnext#14930]
I have set up a loyalty program with two tiers.

The upper tier has a minimum spend of 100,000

I assume that the customer needs to spend at least 100,000 before he moves up to the second tier.

The system is not behaving this way, the system automatically starts applying the second tier from the second invoice regardless of the minimum spend by the customer.

I believe this is a bug that needs to be addressed.

Olamide

**Comment by Zlash65** [comment]
The minimum spent is actually not a single purchase related. Basically tier is calculated based on the transaction done by the customer during the applicable period of the assigned Loyalty Program.

So, lets say - Customer 'A' is assigned Loyalty Program 'LP1'. LP1 has two tiers namely (Tier1 where min_spent is 40k) and (Tier2 where min_spent is 100k). So firstly, A did a transaction of 45k, which means he passed the eligibility for Tier1 and accordingly that tier gets assigned to him. Now he does two transaction of 25k each and gets the benefit of tier1 still as he already got eligible for it. The total transaction done so far is 95k. Now if A performs another transaction of 10k, his total expenditure so far crosses the 100k limit and he gets upgraded to Tier2.

So, to sum it up, the Tier is not based on a single transaction threshold, but rather a sum of transactions during that period of assigned Loyalty Program.

**Comment by olamide2** [comment]
Hello @Zlash65 

You have explained how the system should operate. I agree with you.

But that is not how the system operates. Once you have two tiers, the system automatically takes the customer to the next tier even when the cumulative spend is not yet up to the minimum for the next tier.

Please check this out and confirm this bug.

Regards

**Comment by olamide2** [comment]
Hello @manassolanki 

This issue still persists.

If i have two tiers and the minimum spend for tier 1 is 0 and the minimum spend for tier 2 is 20,000.

Regardless of the total spend by the customer the system applies the loyalty for the upper tier, even if the total spend is 5,000 which is lower than the minimum.

Can you please look into this? As it stands this makes the loyalty program useless.

@rmehta sorry for tagging you but it looks like this issue is not been addressed, I raised it months ago

**Comment by rmehta** [comment]
@olamide2 why don't you get on a support contract with someone so these issues get fixed when you want this. You cannot force anyone to volunteer to fix any issue on priority.

**Comment by olamide2** [comment]
@rmehta,

I am just doing my part. Version 11 is a new ERP version. According to your several write ups, one of the ways to contribute to ERPNext is to text new features and report bugs.

This I am doing, the reason I keep bringing this up is because no one has confirmed that this is a bug. Once this is confirmed as a bug, no pressure.

I am just contributing my bit, the way I can. At the moment I am only using V11 for testing.

Olamide

**Comment by olamide2** [comment]
And by the way, I do have developers on contract now, that is why you may have noticed that I do not bring issues on V 10 to any of the forums, we just fix them, and we also made pull requests for some that are generic.

But v 11 is still under development and I felt the team will appreciate testers like me bringing up issues for resolution before releasing.

Regards

**Comment by Shirkit** [comment]
The issue is still present in v15, where I have a basic tier with 1,00 mininum spend, and a starting tier with 400,00 minimum spend. If the client does a purchase (any value between 1 and 399), he goes straight to the starting tier and skipping the basic tier.
